### ISSUE TITLE
Pipeline is very slow

### ISSUE THREAD

**Issue by Mohamed209** [ifzhang/ByteTrack#42]
check my log.txt file attached , got 0.9 FPS , 11 seconds demo video (nearly 300 frames) may take around 5-7 minutes to fininsh
2021-11-01 14:01:50.434 | INFO     | __main__:main:290 - Args: Namespace(camid=0, ckpt='pretrained/bytetrack_x_mot17.pth.tar', conf=None, demo='video', device='gpu', exp_file='exps/example/mot/yolox_x_mix_det.py', experiment_name='yolox_x_mix_det', fp16=True, fuse=True, match_thresh=0.8, min_box_area=10, mot20=False, name=None, nms=None, path='./videos/palace.mp4', save_result=True, track_buffer=30, track_thresh=0.5, trt=False, tsize=None)
/home/mossad/projects/ByteTrack/venv/lib/python3.8/site-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
2021-11-01 14:01:51.169 | INFO     | __main__:main:300 - Model Summary: Params: 99.00M, Gflops: 791.73
2021-11-01 14:01:53.548 | INFO     | __main__:main:311 - loading checkpoint
2021-11-01 14:02:00.999 | INFO     | __main__:main:315 - loaded checkpoint done.
2021-11-01 14:02:00.999 | INFO     | __main__:main:318 - 	Fusing model...
/home/mossad/projects/ByteTrack/venv/lib/python3.8/site-packages/torch/nn/modules/module.py:561: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the gradient for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more information.
  if param.grad is not None:
2021-11-01 14:02:01.587 | INFO     | __main__:imageflow_demo:236 - video save_path is ./YOLOX_outputs/yolox_x_mix_det/track_vis/2021_11_01_14_02_01/palace.mp4
2021-11-01 14:02:01.589 | INFO     | __main__:imageflow_demo:246 - Processing frame 0 (100000.00 fps)
2021-11-01 14:02:24.684 | INFO     | __main__:imageflow_demo:246 - Processing frame 20 (0.92 fps)
2021-11-01 14:02:47.751 | INFO     | __main__:imageflow_demo:246 - Processing frame 40 (0.92 fps)
2021-11-01 14:03:11.332 | INFO     | __main__:imageflow_demo:246 - Processing frame 60 (0.91 fps)
2021-11-01 14:03:35.619 | INFO     | __main__:imageflow_demo:246 - Processing frame 80 (0.90 fps)
2021-11-01 14:03:58.876 | INFO     | __main__:imageflow_demo:246 - Processing frame 100 (0.90 fps)
2021-11-01 14:04:21.899 | INFO     | __main__:imageflow_demo:246 - Processing frame 120 (0.90 fps)
2021-11-01 14:04:44.870 | INFO     | __main__:imageflow_demo:246 - Processing frame 140 (0.91 fps)

**Comment by AhmedKhaled945** [comment]
Deploying it as ONNX (manually or using the scripts in the repo) would make it a lot faster, not real time but at least 15 FPS.
Maybe with quantization you can achieve a 20 FPS (Near real time performance)

**Comment by hongsamvo** [comment]
I used docker and demo guide from author and I can achieve 21 fps with 2080ti

**Comment by AydinAlptug** [comment]
> Deploying it as ONNX (manually or using the scripts in the repo) would make it a lot faster, not real time but at least 15 FPS. Maybe with quantization you can achieve a 20 FPS (Near real time performance)

Can you please explain the quantization a bit? Does it possible to do while working with data from rtsp?

**Comment by iamrajee** [comment]
@Mohamed209 Did you manage to resolve this?

**Comment by Mohamed209** [comment]
low fps is due to yolox detector , i entegrated Byte track with yolov5 instead of yolox , got accepted fps after converting yolov5 to tensorrt

**Comment by AydinAlptug** [comment]
> low fps is due to yolox detector , i entegrated Byte track with yolov5 instead of yolox , got accepted fps after converting yolov5 to tensorrt

According to your comment, I would like to test converting yolov5 into onnx and test it with ByteTrack. I would be happy if you help me with how to generate the **exp** file [like these](https://github.com/ifzhang/ByteTrack/tree/main/exps/example/mot). 

Also, can you share the changes you made to 
 integrate the yolov5 with ByteTrack.
